SEP-6 & 24: Add `features` object to GET /info (#1043)
stellar/stellar-protocol: ecosystem/sep-0006.md

**Anchors**: To support claimable balances anchors must

> * Set the `claimable_balances` attribute within `GET /info`'s `feature_flags` object to `true`

Reviewer: Is there any negative affect on the situation where anchors have already implemented claimable balances but not the feature flags object?

I'm wondering if a client could be surprised, thinking that claimable balances would not be used because the default is false. 

Would it work to say that a missing features flag object means the client doesn't know what they support?

Author: Today, all wallets have to assume anchors do not support claimable balances, so I don't think this will be a problem.

If a wallet _knows_ an anchor supports claimable balances despite not yet updating their `GET /info` response, it can leave it's implementation unchanged -- passing `claimable_balance_supported: true` in the `GET /deposit` request and waiting for the claimable balance to be created. Or, it could implement the `features` detection while also adding an allowlist of sorts simultaneously. Its not perfect but I think the process is manageable.

  "transaction": {
    "enabled": false,
    "authentication_required": true
  },

>   "feature_flags": {

Reviewer: We could also name this `features`.

Author: I think thats better 👍

* `enabled`: `true` if the endpoint is available.
* `authentication_required`: `true` if client must be [authenticated](#authentication) before accessing the endpoint.

#### Feature Flags

The `feature_flags` object contains boolean values indicating whether or not specific features are supported by the anchor. If the object or specific feature is not present in the response, the default value described below may be assumed. This information enables wallets to adjust their behavior based on the feature set supported by the anchor.

Name | Default | Description
-----|---------|------------
`account_creation` | `true` | Whether or not the anchor supports creating accounts for users requesting deposits.
`claimable_balances` | `false` | Whether or not the anchor supports sending deposit funds as claimable balances. This is relevant for users of Stellar accounts without a trustline to the requested asset.

> The default values for the features listed above have been selected based on the ecosystem's current support. It is highly recommened to support all features enumerated for the best user experience.

Reviewer: I don't think we need to say it's recommended, which has the spec offer opinion, rather we can focus on highlighting how to get the best use experience. 

```suggestion
The default values for the features listed above have been selected based on the ecosystem's current support. Supporting all features provided the best user experience.
```
